tests/functional/warehouse_test/test_warehouses.py

```diff
@@ -90,3 +89,25 @@ def test_snowflake_override_ok(self, project):
             ]
         )
         check_relations_equal(project.adapter, ["OVERRIDE_WAREHOUSE", "EXPECTED_WAREHOUSE"])
+
+
+class TestInvalidConfigWarehouse:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "invalid_warehouse.sql": project_config_models__invalid_warehouse_sql,
+        }
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "config-version": 2,
+            "models": {
+                "test": {
+                    "snowflake_warehouse": "DBT_TEST_DOES_NOT_EXIST",
```

Comment:
Yes, nicely done - we should expect the dbt test to try using a Snowflake warehouse that doesn't exist (`use warehouse DBT_TEST_DOES_NOT_EXIST`), where previously it wouldn't, and so it should fail with `Object does not exist, or operation cannot be performed.`

Reply:
went ahead and also added a assert to catch if that error message stops showing up.

Reply:
For the sake of completeness, can we add a positive test case that uses this config option, just to make sure it's setting it? The error message we're getting in the negative test case is pretty general.

Reply:
added a new test